src/entities/userFactory.ts

```diff
@@ -0,0 +1,10 @@
+import {Injectable} from '@nestjs/common';
+import {User} from './user';
+import {UserData} from './user.interface';
+
+@Injectable()
+export class UserFactory {
+    create(data: UserData) {
+        return new User(data);
+    }
+}
```

Comment:
I suggest to use so-called "factory method"
This approach allows us to "inject" factory inside entity. 
To achieve it do:
```
User {
    private constructor() {}
    public static create() { 
        return new User();
    }
 }
```